### ISSUE TITLE
Would you accept a PR for .tap ?

### ISSUE THREAD

**Issue by FGRibreau** [mwilliamson/node-options#13]
:)

**Comment by mwilliamson** [comment]
As in a function that behaves as 

```javascript
const tap = (option, f) => option.map(value => (f(value), value));
```

? Could you give some examples of when this might be useful?

**Comment by FGRibreau** [comment]
Exactly :)

It's useful for mutation (sometimes needed) and debugging purpose.

e.g.: 

```js
option.tap(val => val.x += 1).valueOrElse(0);
```

**Comment by mwilliamson** [comment]
My preference is to use map in a separate statement. This keeps the API smaller, and makes the flow of data clearer without having to understand `tap`.

```javascript
option.map(val => val.x += 1);
const value = option.valueOrElse(0);
```

**Comment by FGRibreau** [comment]
Indeed, one could always mimic .tap behaviour using map but my request concerns code flow & conciseness, it's not a must have but a very good nice to have and every time I use this library, I miss it :)
